### PR TITLE
Edit event

### PR DESCRIPTION
**Pull request by dieunb**

### CODE REVIEW

File: app/controllers/api/events_controller.rb [hunk 1]
```diff
@@ -47,11 +47,23 @@ def update
 
   def show
     @event = Event.find_by id: params[:id]
+
+    locals = {
+            start_date: params[:start],
```
Comment (nhunght) [inline app/controllers/api/events_controller.rb]: indent ạ

Comment (nhunght) [inline app/controllers/api/events_controller.rb]: indent

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -68,14 +71,38 @@ def create
...
+    Notification.all.each do |notification|
+      @event.notification_events.find_or_initialize_by notification: notification
+    end
+
+    RepeatOn.repeat_ons.values.each do |repeat_on|
+      @event.repeat_ons.find_or_initialize_by repeat_on: repeat_on
+    end
+
+    data = JSON.parse Base64.urlsafe_decode64(params[:fdata])
+    @event.start_date = DateTime.strptime(data["start_date"], "%m-%d-%Y %H:%M %p")
```
Comment (tqtrung09) [inline app/controllers/events_controller.rb]: Move format date to Setting.yml

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -68,14 +71,38 @@ def create
...
+    data = JSON.parse Base64.urlsafe_decode64(params[:fdata])
+    start_date = DateTime.strptime(data["start_date"], "%m-%d-%Y %H:%M %p")
+    finish_date = DateTime.strptime(data["finish_date"], "%m-%d-%Y %H:%M %p")
+
+    if @event.start_date == start_date
+      @event.assign_attributes event_params
+    else
+      @event = Event.new event_params.merge({parent_id: @event.id})
+      @event.exception_time = @event.start_date
+      @event.exception_type = "edit_only"
```
Comment (tqtrung09) [inline app/controllers/events_controller.rb]: Event.exception_types[:edit_only]

**Pull request merged**